common/src/lib.rs

```diff
@@ -165,11 +165,8 @@ pub static WARP_CMD_CH: Lazy<WarpCmdChannels> = Lazy::new(|| {
 // allows the UI to receive events to Warp
 // pretty sure the rx channel needs to be in a mutex in order for it to be a static mutable variable
 pub static WARP_EVENT_CH: Lazy<WarpEventChannels> = Lazy::new(|| {
-    let (tx, rx) = tokio::sync::mpsc::unbounded_channel();
-    WarpEventChannels {
-        tx,
-        rx: Arc::new(Mutex::new(rx)),
-    }
+    let (tx, _) = broadcast::channel(128);
```

Comment:
Interesting to use broadcast here. I dont know the amount of events that are emitted at one time in uplink but keep in mind of the amount in case of an overflow. It is better to use though instead of anchoring the receiver behind a lock primitive.